tests/tests.rs

```diff
@@ -48,6 +48,41 @@ fn test_parse_tree_basics() {
     assert_resource_equal(a, resource);
 }
 
+#[test]
+fn test_basic_parse_tree_with_condition() {
+    let a: Value = serde_json::json!({
+        "LogicalResource": {
+            "Type": "AWS::IAM::Role",
+            "Condition": "SomeCondition",
+            "Properties": {
+                "RoleName": "bob",
+                "AssumeTime": 20,
+                "Bool": true,
+                "NotExistent": {"Ref": "AWS::NoValue"},
+                "Array": ["hi", "there"]
+            }
+        }
+    });
+
+    let resource = ResourceParseTree {
+        name: "LogicalResource".into(),
+        condition: Option::Some("SomeCondition".into()),
+        metadata: Option::None,
+        update_policy: Option::None,
+        deletion_policy: Option::None,
+        dependencies: vec![],
+        resource_type: "AWS::IAM::Role".into(),
+        properties: map! {
+            "RoleName" => ResourceValue::String("bob".into()),
+            "AssumeTime" => ResourceValue::Number(20),
+            "Bool" => ResourceValue::Bool(true),
+            "NotExistent" => ResourceValue::Null,
+            "Array" => ResourceValue::Array(vec![ResourceValue::String("hi".into()), ResourceValue::String("there".into())])
+        },
+    };
+    assert_resource_equal(a, resource);
```

Comment:
Me reading this: "why a as a variable name?"
Past me: "a is dope as hell", according to the rest of the test suite 😂

It's fine, will add a pr to rename this later :)

Reply:
Yeah lol. I was like "why are these all a? ... whatever I'll just stick with the way he's been doing it